tests/testthat/test-3agglomerate.R

```diff
@@ -91,20 +91,20 @@ test_that("agglomerate", {
     data(SilvermanAGutData)
     se <- SilvermanAGutData
     # checking reference consensus sequence generation
-    actual <- mergeFeaturesByRank(se,"Genus", mergeRefSeq = FALSE)
+    actual <- mergeFeaturesByRank(se,"Genus", mergeRefSeq = FALSE, onRankOnly = TRUE)
     expect_equal(as.character(referenceSeq(actual)[[1]]),
-                 paste0("TCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGTTTGATAA",
-                        "GTTAGAGGTGAAATCCCGGGGCTTAACTCCGGAACTGCCTCTAATACTGTTAG",
-                        "ACTAGAGAGTAGTTGCGGTAGGCGGAATGTATGGTGTAGCGGTGAAATGCTTA",
-                        "GAGATCATACAGAACACCGATTGCGAAGGCAGCTTACCAAACTATATCTGACG",
-                        "TTGAGGCACGAAAGCGTGGGG"))
+                 paste0("GCAAGCGTTAATCGGAATTACTGGGCGTAAAGCGCACGCAGGCGGTCTGTCA",
+                        "AGTCGGATGTGAAATCCCCGGGCTCAACCTGGGAACTGCATTCGAAACTGGC",
+                        "AGGCTAGAGTCTTGTAGAGGGGGGTAGAATTCCAGGTGTAGCGGTGAAATGC",
+                        "GTAGAGATCTGGAGGAATACCGGTGGCGAAGGCGGCCCCCTGGACAAAGACT",
+                        "GACGCTCAGGTGCGAAAGCGTGGGG"))
     actual <- mergeFeaturesByRank(se,"Genus", mergeRefSeq = TRUE)
     expect_equal(as.character(referenceSeq(actual)[[1]]),
-                 paste0("BCNMKCKTTVWYCKKMHTTMYTKKKYKTMMMKNKHDYKYMKDYKKNHNNNYMM",
-                        "KHHNDNNKTKMMMDNBHNBKKCTYMMCHNBNDDDNKSSHBNNRWDMYKKBNND",
-                        "NYTDRRKDVHNKNDRVGRNDRSBRRAWTBYNHRKKKWRSSRKKRAAWKSSKWR",
-                        "RWDWTNDBRVRRAMHHCMRDKKSSRARGSSVSYYHNYBRRVHNDNNHYKRMVV",
-                        "YKVRDNNNSRAARSBDKGGKK"))
+                 paste0("GCGAGCGTTATCCGGAATCATTGGGCGTAAAGGGTGCGTAGGCGGCGAGTTA",
+                        "AGTCTGAGGTAAAAGGTTGCAGCTCAACTGTAACAAGCCTTGGAAACTGACT",
+                        "AGCTAGAGTGCAGGAGAGGGCAGTGGAATTCCATGTGTAGCGGTAAAATGCG",
+                        "TAGATATATGGAGGAACACCAGTGGCGAAGGCGGCTGTCTGGCCTGTAACTG",
+                        "ACGCTGAGGCACGAAAGCGTGGGG"))
```

Comment:
This result has changed? It should not be since the earlier onRankOnly default value was FALSE.

Probably if you use na.rm = FALSE it works with these sequences

Reply:
We had changed onRankOnly default to TRUE when we did the critical fix. I didn't catch the error back then since i hadn't miaTime. I will checkout the commit before the change and see if the sequence was valid then.

The sequence with na.rm TRUE/FALSE is also different.

Reply:
I think when we  ordered in alphabetical order it changed the sequence.

Reply:
Instead of taking the first one, can you take one (by name) which is the correct taxon?

You can debug the function and check what is the first one before the sorting

Reply:
the first string before sorting is the older sequence.

```
> data(SilvermanAGutData)
> se <- SilvermanAGutData
> # checking reference consensus sequence generation
> actual <- mergeFeaturesByRank(se,"Genus", mergeRefSeq = FALSE)
Called from: .local(x, ...)
Warning message:
In mergeFeaturesByRank(se, "Genus", mergeRefSeq = FALSE) :
  'mergeFeaturesByRank' is deprecated. Use 'agglomerateByRank' instead.
Browse[1]> q <- as.character(referenceSeq(x)[[1]])
Browse[1]> q
[1] "TCAAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGTTTGATAAGTTAGAGGTGAAATCCCGGGGCTTAACTCCGGAACTGCCTCTAATACTGTTAGACTAGAGAGTAGTTGCGGTAGGCGGAATGTATGGTGTAGCGGTGAAATGCTTAGAGATCATACAGAACACCGATTGCGAAGGCAGCTTACCAAACTATATCTGACGTTGAGGCACGAAAGCGTGGGG"
Browse[1]> 
```

Reply:
Check the name of it and get its sequence in tests (instead of taking first one)